### DISCUSSION TITLE
repository releases, tags and edit

### DISCUSSION THREAD

**Discussion by nichitapavel** [gogs/gogs#6945] [Q&A] [answered]
Hello.
In gogs when I add a tag to a repository it will automatically create a release with the **.tar.gz** and **.zip** archives.
The questions are:
- Where are these archives stored in gogs server? Are the created on the fly when downloading them?
- How can I add a binary (or any other thing) to a release created by a git tag? If I create a release from the page I can add files, but I do not see this option for a release created by a git tag.

INFO:
**Gogs version**  0.12.3
**Git version**   2.25.1
**Go version**  go1.15
**Build time**     2020-10-07 03:03:48 UTC
**Build commit**   f0e3cd90f8d7695960eeef2e4e54b2e717302f6c

**Comment by unknwon** [comment, marked as answer]
> * Are the created on the fly when downloading them?

Yes, they are always computed on the fly, provided by the Git CLI.



> * How can I add a binary (or any other thing) to a release created by a git tag? If I create a release from the page I can add files, but I do not see this option for a release created by a git tag.

Yes, you will need to create a release in order to upload files, that's how Gogs keep track of things.
